include/bm/bm_sim/extern.h

```diff
@@ -52,6 +52,8 @@ class ExternFactoryMap {
   std::unordered_map<std::string, ExternFactoryFn> factory_map{};
 };
 
+#define TO_STRING(name) #name
```

Comment:
Please rename to something less generic to avoid name collisions, maybe `_BM_EXTERN_TO_STRING`